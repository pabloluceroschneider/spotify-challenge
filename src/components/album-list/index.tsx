import Link from 'next/link';
import Image from 'next/image';

import { Albums } from '@/types/spotify';

import styles from './styles.module.css';
import { FC } from 'react';

interface Props {
  data: Albums;
}

export const AlbumList: FC<Props> = ({ data }) => {
  return (
    <section className={`${styles.search}`}>
      <div className={`${styles.results}`}>
        {data?.items?.map((album: any) => (
          <Link key={album.id} href={`album/${album.id}`}>
            <div className={`${styles.album}`} title={album.name}>
              <div className={`${styles['album-details']}`}>
                <h3 className={`${styles['album-name']}`}>{album.name}</h3>
                <p>{album.artists[0].name}</p>
              </div>
              <Image
                className={`${styles['album-img']}`}
                src={album.images[0].url}
                alt={album.name}
                fill
              />
            </div>
          </Link>
        ))}
      </div>
    </section>
  );
};
