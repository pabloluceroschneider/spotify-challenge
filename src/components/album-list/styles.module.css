.search {
  width: 64rem;
  display: flex;
}

.results {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 3rem;
}

.album {
  min-width: 0;
  position: relative;
  height: 14rem;
}

.album-img {
  object-fit: cover;
}

.album-details {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  padding: 0.5rem;
  background: rgba(0, 0, 0, 0.8);
  color: white;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  gap: 0.25rem;
  text-align: right;
}

.album-name {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
