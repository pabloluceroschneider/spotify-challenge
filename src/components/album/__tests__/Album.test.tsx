/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Album } from '..';
import { data } from '@/mocks/album-list.mock';

describe('Album', () => {
  let props = { ...data };

  it('renders an album', () => {
    const { baseElement } = render(<Album {...props} />);

    expect(baseElement).toMatchSnapshot();
  });
});
