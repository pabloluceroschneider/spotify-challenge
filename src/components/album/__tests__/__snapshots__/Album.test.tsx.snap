// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders an album 1`] = `
<body>
  <div>
    <a
      href="album/0mEY1U7SZdIwDWylFm2UC4"
      id="0mEY1U7SZdIwDWylFm2UC4"
    >
      <div
        class="album"
        title="Doble Vida (Remastered)"
      >
        <div
          class="album-details"
        >
          <h3
            class="album-name"
          >
            Doble Vida (Remastered)
          </h3>
          <p>
            Soda Stereo
          </p>
        </div>
        <img
          alt="Doble Vida (Remastered)"
          class="album-img"
          data-nimg="1"
          decoding="async"
          height="640"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.scdn.co%2Fimage%2Fab67616d0000b2733d92b2ad5af9fbc8637425f0&w=3840&q=75 3840w"
          style="color: transparent; width: 100%; height: auto;"
          width="640"
        />
      </div>
    </a>
  </div>
</body>
`;
